### DISCUSSION TITLE
blade docblocks suggestion

### DISCUSSION THREAD

**Discussion by fcolecumberri** [laravel/framework#37593] [Ideas]
I have been thinking about creating a package to generate documentation for blade/livewire components (maybe already exists, but I have not found one) and using the docblocks th the php files. the only problem I find is that there are [Anonymous Components](https://laravel.com/docs/8.x/blade#anonymous-components) that may use the `@props` directive, so I was wondering if someone has thought about a blade docblock directive, I was thinking about adding an extra `-` to the `{{--` (similar to the extra `*` in the `/**` classic docblock) so using the [documentation example](https://laravel.com/docs/8.x/blade#data-properties-attributes) the idea would be something like this:

```blade
{{---
 - @param message description
--}}
@props(['type' => 'info', 'message'])

<div {{ $attributes->merge(['class' => 'alert alert-'.$type]) }}>
    {{ $message }}
</div>
```

But I was wondering if maybe this is already done.

(Also maybe this is not the place to discuss this, if there is a better place, tell me)

**Comment by Wulfheart** [comment]
Have a look at https://gummibeer.dev/blog/2020/phpdoc-in-blade-views/. @Gummibeer wrote a great blog post about this.

**Reply by Gummibeer** [reply]
Thanks Alex! 🙏
The problem with a custom doc-block syntax would be that no IDE will understand it and it will take time, if ever, till they add the new syntax as well.
Doc-blocks without IDE support are kind of useless. 😟
